src/prefect/engine.py

```diff
@@ -1861,14 +1861,35 @@ def resolve_input(expr, context):
 
         return result_by_state.get(state)
 
-    return visit_collection(
-        parameters,
-        visit_fn=resolve_input,
-        return_data=return_data,
-        max_depth=max_depth,
-        remove_annotations=True,
-        context={},
-    )
+    if isinstance(parameters, dict):
```

Comment:
@madkinsz `parameters` sometimes seems to be called like `resolve_inputs(parameters=wait_for, ...)` where `wait_for` is a `Iterable[PrefectFuture]`. Is this an okay solution to conditionally do this?

Reply:
Thanks for looking into this!

We should probably do `resolve_inputs(parameters={"wait_for": wait_for})` instead since that key is reserved anyway it'll be helpful for understanding where failures come from in that case as well.

Reply:
ah got it! thanks will update and hopefully CI should clear up 🤞